crates/bevy_ecs/src/query/default_filters.rs

```diff
@@ -0,0 +1,173 @@
+use crate as bevy_ecs;
+use crate::{component::ComponentId, query::FilteredAccess};
+use bevy_ecs_macros::Resource;
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+enum FilterKind {
+    With,
+    Without,
+}
+
+#[derive(PartialEq, Eq, Debug)]
+struct DefaultFilter {
+    component_id: ComponentId,
+    kind: FilterKind,
+}
+
+/// A list of default query filters, these can be used to globally exclude entities from queries.
+/// Each individual filter is only applied to queries that don't mention that component.
+///
+/// If for example we register a `Hidden` component using the `without` method, entities with this
+/// component will only be visible to a [`Query`](crate::prelude::Query) containing something like
+/// [`With<Hidden>`](crate::prelude::With) or [`Has<Hidden>`](crate::prelude::Has).
+///
+/// These filters are only applied to queries initialized after updating this resource,
+/// it should most likely only be modified before the app starts.
+#[derive(Resource, Default)]
+pub struct DefaultQueryFilters(Vec<DefaultFilter>);
+
+impl DefaultQueryFilters {
+    /// Add a With filter to the default query filters.
+    /// Removes any Without filter for this component if present.
+    pub fn with(&mut self, component_id: ComponentId) {
```

Comment:
These methods should be renamed to `with_untyped` etc for consistency.

Then, I think we should add methods on `World` for `default_with_filter` etc that take a type.